audiomentations/augmentations/room_simulator.py

```diff
@@ -0,0 +1,355 @@
+import sys
+from typing import Optional
+import numpy as np
+import random
+
+from audiomentations.core.transforms_interface import BaseWaveformTransform
+
+
+class RoomSimulator(BaseWaveformTransform):
+    """
+    A ShoeBox Room Simulator. Simulates a cuboid of parametrized size and 
+    average surface absorption coefficient. It also includes a source \
+    and microphones in parametrized locations.
+
+    Use it when you want a ton of synthetic room impulse responses of specific configurations
+    characteristics or simply to quickly add reverb for augmentation purposes
+
+    Some examples:
+
+    > augment = RoomSimulator(use_ray_tracing=False)
+    Quickly add reverberation without caring for accuracy.
+
+    > augment = RoomSimulator(calculate_by_absorption_or_rt60="rt60", min_absorption_value_or_rt60=0.06, max_absorption_value_or_rt60=0.06, min_size_x = ...)
+    Augment with randomly selected room impulse responses that have an RT60 of 0.06.
+
+    > augment = RoomSimulator(min_mic_radius=1.0, max_min_radius=1.0)
+    Augment with a RIR captured by all positions of the microphone on a sphere, centred around the source at 1m
+
+    > augment = RoomSimulator(min_mic_radius=1.0, max_min_radius=1.0, min_mic_elevation=0.0, max_mic_elevation=0.0)
+    Augment with a RIR captured by all positions of the microphone on a circle, centred around the source at 1m    
+    """
+
+    supports_multichannel = True
+
+    def __init__(
+        self,
+        min_size_x: float = 1.0,
+        max_size_x: float = 1.0,
+        min_size_y: float = 1.0,
+        max_size_y: float = 1.0,
+        min_size_z: float = 1.0,
+        max_size_z: float = 1.0,
```

Comment:
Why are we using a single fixed-size room by default? My first thought is that it makes sense to have various random room sizes. Does the usage of this depend on the value of calculate_by_absorption_or_rt60?